lib/Frontend/nvqpp/ConvertStmt.cpp

```diff
@@ -130,10 +130,28 @@ bool QuakeBridgeVisitor::TraverseCXXForRangeStmt(clang::CXXForRangeStmt *x,
   assert(stdvecTy && "expected a std::vector<T>");
   auto eleTy = stdvecTy.getElementType();
   auto i64Ty = builder.getI64Type();
-  auto iters = builder.create<cc::StdvecSizeOp>(loc, i64Ty, buffer);
   auto dataPtrTy = cc::PointerType::get(eleTy);
   auto dataArrPtrTy = cc::PointerType::get(cc::ArrayType::get(eleTy));
-  auto ptr = builder.create<cc::StdvecDataOp>(loc, dataArrPtrTy, buffer);
+  auto [iters, ptr] = [&]() -> std::pair<Value, Value> {
+    if (auto call = buffer.getDefiningOp<func::CallOp>())
+      if (call.getCallee().equals(setCudaqRangeVector)) {
```

Comment:
Note that this all works fine if these two lines are removed. So this peephole optimization isn't required. But it is nice. :)